autoload/neomake/highlights.vim

```diff
@@ -63,7 +63,12 @@ function! neomake#highlights#AddHighlight(entry, type) abort
     let l:hi = get(s:highlight_types, toupper(a:entry.type), 'NeomakeError')
     if get(g:, 'neomake_highlight_lines', 0)
         if s:nvim_api
-            call nvim_buf_add_highlight(a:entry.bufnr, s:highlights[a:type][a:entry.bufnr], l:hi, a:entry.lnum - 1, 0, -1)
+            " some makers use line 0 for file warnings
+            " (e.g. cpplint with no copyright warnings)
+            " these cannot be highlighted since the line does not exist
+            if a:entry.lnum > 0
+                call nvim_buf_add_highlight(a:entry.bufnr, s:highlights[a:type][a:entry.bufnr], l:hi, a:entry.lnum - 1, 0, -1)
+            endif
```

Comment:
I think this should be skipped in general (at the top of the function) instead.
Or even with the caller(s) of the function?!

side note: often (with other plugins etc) line numbers < 1 get normalized to 1, but I think it makes sense to have something like global items (with line number 0).